giskard/llm/evaluators/base.py

```diff
@@ -1,45 +1,54 @@
-from typing import Any, Dict, Optional, Sequence
+from typing import Dict, Optional, Sequence, Tuple
 
+import json
+import logging
 from abc import ABC, abstractmethod
-from dataclasses import dataclass
+from dataclasses import dataclass, field
 
-from ...core.test_result import TestResultDetails, TestResultStatus
+from ...core.test_result import TestResultStatus
 from ...datasets.base import Dataset
 from ...models.base.model import BaseModel
 from ..client import LLMClient, get_default_client
+from ..client.base import ChatMessage
 from ..errors import LLMGenerationError
+from .utils import format_conversation
 
-EVALUATE_MODEL_FUNCTIONS = [
-    {
-        "type": "function",
-        "function": {
-            "name": "evaluate_model",
-            "description": "Evaluates if the model passes the test",
-            "parameters": {
-                "type": "object",
-                "properties": {
-                    "passed_test": {
-                        "type": "boolean",
-                        "description": "true if the model successfully passes the test",
-                    },
-                    "reason": {
-                        "type": "string",
-                        "description": "optional short description of why the model does not pass the test, in 1 or 2 short sentences",
-                    },
-                },
-                "required": ["passed_test"],
-            },
-        },
-    },
-]
+logger = logging.getLogger("giskard.llm")
+
+
+@dataclass
+class EvaluationResultExample:
+    # The status of the example
+    status: TestResultStatus
+    sample: Optional[Sequence[Dict]]
+    # The reason why the example have given status
+    reason: Optional[str] = None
+
+    def to_example(self):
+        if self.status == TestResultStatus.ERROR:
+            return {"error": self.reason, "sample": self.sample}
+        else:
+            return {"reason": self.reason, "sample": self.sample}
+
+    def to_dict(self):
+        return {"reason": self.reason, "sample": self.sample, "status": self.status}
 
 
 @dataclass
 class EvaluationResult:
-    failure_examples: Sequence[dict]
-    success_examples: Sequence[dict]
-    errors: Sequence[dict]
-    details: Optional[TestResultDetails] = None
+    results: Sequence[EvaluationResultExample] = field(default_factory=list)
```

Comment:
since we're calling `.append` below this should rather be a List, not a Sequence